sympy/sets/fancysets.py

```diff
@@ -472,14 +469,20 @@ def __new__(cls, *args):
         if start.is_infinite:
             end = stop
         else:
-            ref = start if start.is_finite else stop
-            n = ceiling((stop - ref)/step)
-            if n <= 0:
-                # null Range
-                start = end = S.Zero
-                step = S.One
+            if not all(w.is_number for w in [start, stop, step]):
+                end = stop
             else:
-                end = ref + n*step
+                ref = start if start.is_finite else stop
+                if any(i.is_number and i.has(Float) for i in (step, stop - ref)):
```

Comment:
This will still allow `Range(Float(1), Float(1) + a)` but there is no `a` that you can add other than 0 that will give anything but a Float for `stop - ref`. And that case will give you a null range anyway...so how 
```python
if Tuple(ref, stop, step).has(Float):
    raise ValueError('Floats not allowed in Range')
```

Reply:
This also still allows `1 + sqrt(2)` in the args, too. What do you need for the stats? Do you need `pi`? Otherwise let's keep the args at Integer or Symbol. But do you need `f(x)`? It would be good to see an example of how this is going to be used.

Reply:
I only need `Symbol`. I think we should take a stand and only `Symbol` and `Integer`. Infact, any expression should also contain only `Symbol` and `Integer`. Apart from that everything should be rejected as it would cause a lot of chaos later on. What do you say?